app/src/components/Map/SearchPanel.jsx

```diff
@@ -1,84 +1,108 @@
 import React, { Component } from 'react';
-import FiltersPanel from '../../containers/Map/FilterPanel';
-import controlPanelStyle from '../../../styles/components/c-control_panel.scss';
-import { Accordion, AccordionItem } from 'react-sanfona';
+import classnames from 'classnames';
+
+import SearchResult from './SearchResult';
+import iconsStyles from '../../../styles/icons.scss';
+import searchPanelStyles from '../../../styles/components/map/c-search-panel.scss';
 
 class SearchPanel extends Component {
 
   constructor(props) {
     super(props);
 
     this.state = {
-      visibleAdvancedSearch: false
+      isEmpty: true,
+      search: this.props.search
     };
+  }
 
-    this.toggleVisibleAdvancedSearch = this.toggleVisibleAdvancedSearch.bind(this);
-    this.onSearchInputChange = this.onSearchInputChange.bind(this);
+  componentWillReceiveProps(nextProps) {
+    if (!!nextProps.search) {
+      Object.assign(this.state.search, nextProps.search);
```

Comment:
Nope, nope, nope. You're already mutating the state here. Because you only want to update the `search` property, just do that instead:
```javascript
this.setState({ search: nextProps.search });
```